src/sagas/combat-sagas.tsx

```diff
@@ -0,0 +1,193 @@
+import { takeEvery, put, all, select } from "redux-saga/effects";
+import {
+  ENTER_COMBAT,
+  CombatAction,
+  SELECT_PLAYER_COMBAT_ACTION,
+  SELECT_ENEMY_COMBAT_ACTION,
+  SelectEnemyCombatActionAction,
+  CombatActionType,
+  AlterEnemyHealthAction,
+  ALTER_ENEMY_HEALTH,
+  AlterMomentumAction,
+  ALTER_MOMENTUM
+} from "../player-actions/violence/types";
+import { AddMessageAction, ADD_MESSAGE } from "../message-feed/types";
+import { StoreState } from "../redux-common/store";
+import { enemyCombatantReducerDefaultState } from "../redux-common/default-store-state";
+import _ from "lodash";
+import { AlterHealthAction, ALTER_HEALTH } from "../resources/types";
+
+function* combatActionListener() {
+  yield takeEvery(SELECT_PLAYER_COMBAT_ACTION, resolveCombatTurn);
+}
+
+function* resolveCombatTurn() {
+  yield reportSelectedAction();
+  yield reportEnemyAction();
+  yield calculateDamage();
+  yield selectEnemyAction();
+}
+
+function* calculateDamage() {
+  const playerSelectedCombatAction = yield select((state: StoreState) =>
+    state ? state.playerSelectedCombatAction : null
+  );
+  const enemySelectedCombatAction = yield select((state: StoreState) =>
+    state ? state.enemySelectedCombatAction : null
+  );
+  const momentum = yield select((state: StoreState) =>
+    state ? state.momentum : 0
+  );
+
+  const playerAction: CombatAction = playerSelectedCombatAction;
+  const enemyAction: CombatAction = enemySelectedCombatAction;
```

Comment:
what is this doing ? why can't you just directly use `enemySelectedCombatAction?`